signature-aggregator/api/api.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package api
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/awm-relayer/signature-aggregator/aggregator"
+	"github.com/ava-labs/awm-relayer/types"
+	"github.com/ava-labs/awm-relayer/utils"
+	"go.uber.org/zap"
+)
+
+const (
+	RawMessageAPIPath = "/aggregate-signatures/by-raw-message"
+	defaultQuorumNum  = 67
+)
+
+// Defines a request interface for signature aggregation for a raw unsigned message.
+// Currently a copy of the `ManualWarpMessageRequest` struct in relay_message.go
+type AggregateSignaturesByRawMsgRequest struct {
+	// Required. hex-encoded message, optionally prefixed with "0x".
+	UnsignedMessage string `json:"unsigned-message"`
+	// Optional hex or cb58 encoded signing subnet ID. If omitted will default to the subnetID of the source BlockChain
+	SigningSubnetID string `json:"signing-subnet-id"`
+	// Optional. Integer from 0 to 100 representing the percentage of the quorum that is required to sign the message
+	// defaults to 67 if omitted.
+	QuorumNum uint64 `json:"quorum-num"`
```

Comment:
I left a similar comment elsewhere, but "numerator" only makes sense with a corresponding "denominator". Rather than require the user to know that the implicit denominator is 100, a name like `QuorumPercentage` would be clearer.